tests/test_sync_functions.py

```diff
@@ -62,16 +64,16 @@ def test_check_new_videos(self, mock_resp):
         self.assertEqual(sync_module.motion, {})
         self.assertTrue(sync_module.check_new_videos())
         self.assertEqual(
-            sync_module.last_record["foo"],
-            {"clip": "/foo/bar.mp4", "time": "1990-01-01T00:00:00+00:00"},
+            sync_module.last_records["foo"],
+            [{"clip": "/foo/bar.mp4", "time": "1990-01-01T00:00:00+00:00"}],
         )
         self.assertEqual(sync_module.motion, {"foo": True})
         mock_resp.return_value = {"media": []}
         self.assertTrue(sync_module.check_new_videos())
         self.assertEqual(sync_module.motion, {"foo": False})
         self.assertEqual(
-            sync_module.last_record["foo"],
-            {"clip": "/foo/bar.mp4", "time": "1990-01-01T00:00:00+00:00"},
+            sync_module.last_records["foo"],
+            [],
```

Comment:
I didn't realize this should remain.  I am clearing the `last_records` every refresh.  I interpreted it as the list of records since the last refresh.  I'll make some changes so that the most recent record persists.

Reply:
This is the last thing that I need to address.  Hopefully I can knock that out this week.  But my time is about to become more limited.  I may have to wait until the weekend.

I believe you can test against the most recent change noting that I still need to understand this requirement more.

I viewed the `last_record` / `last_records` as the vehicle to shuttle clips into each individual camera.  And the camera keeps a reference to its last clip.  (And now the camera keeps references to multiple clips so that they won't be overwritten before being downloaded.)

Reply:
I had to jump through some hoops, but I got this test back to way it was.

I still don't understand the purpose of maintaining the last record when there is no motion.  And it looks like it is being cached in two places:  sync module _and_ camera.  It seems to me that letting the camera maintain the cache is easier to reason about, and also just letting the sync module discover new information (like newly available videos).